### ISSUE TITLE
Add data stats to view progress

### ISSUE THREAD

**Issue by minthemiddle** [wakatara/harsh#7]
We're trying to keep harsh as minimal as possible, but...

**Is your feature request related to a problem? Please describe.**
I'm frustrated when I only get a visual representation of my habits' progress and no data. I have to parse the text files with some tool like `q` or create a SQLite database out of it to get some basic stats (current streak, longest streak, days tracked, days successful). 

**Describe the solution you'd like**
`harsh stats` would output something along the lines of:

```
habit A : 100 day streak : 400 totally logged, 800 days tracked
```

**Describe alternatives you've considered**
See above: Using external tools like `q` or `SQlite`

**Comment by wakatara** [comment]
Hey @minthemiddle 

Interesting one. I like it. (though not that relevant for me, personally - I just look at the last 100 days in the visual graph to see how I'm doing but can understand wanting it to be more quantifiable.). 
I actually do not think this would be that hard to add in and I agree, it does add a nice summary to the app.

Lemme take a whirl at it between now and end of the weekend. 
So, to be more specific, you want 
streak: logged: days since started tracking
correct? 

(thanks for the suggestion btw)
Daryl.

**Comment by minthemiddle** [comment]
`streak: logged: days since started tracking`

Exactly.

Example in plain English:
- Habit: Play Piano
- Goal: yes/no
- Logged: Logged 280 days (starting 2021-01-01)
- Success: 260 days (eveything at and after 2021-01-20)
- Streak: 260 days

**Comment by wakatara** [comment]
Btw, I forgot to mention @minthemiddle, if you do have a working implementation of this already with the code base. Please feel free to point me at your repo or even branch it as a PR to be pulled in.

**Comment by minthemiddle** [comment]
@wakatara Hi, sorry, I have no working implementation of this. I am not proficient in Rust either, so I fear I cannot help on the technical side.

**Comment by wakatara** [comment]
@minthemiddle Well, it's written in Golang, so Rust proficiency not required.  =p
Add habit: Learn golang weekly: 7 ... =]

Finally getting a bit of time to take a shot at this. Stand by.

**Comment by wakatara** [comment]
Alrighty @minthemiddle ...

I've got an initial implementation of this going, though it calcs on days rather than chains at the moment (interested in whether you and you think others are more interested in days counted versus chains actually, but this was easy to work up , so was thinking of pushing it as an initial feature and then getting feedback on if people totally hated it or not... scarily it's quite interesting seeing all my habits, chain break, and skip totals vs just the 100 day view as I had questioned the utility of this feature originally when you suggested it (so, I stand corrected). 

How does this look as a starting stab? It takes the entire history of your file (mine only ever go a year anyhow) and calcs the nums for you. 

lemme know as I'd like to push this and a really silly bug I coded in (and someone found #facepalm ) end of this weekend. 

<img width="665" alt="Cursor_and_harsh_go_—_harsh" src="https://user-images.githubusercontent.com/2866/126033179-155ea211-d2d8-4f02-99a0-395d83c49928.png">

**Comment by wakatara** [comment]
Actually, @minthemiddle  
this one looks a teeny bit better (and covers the logical error I made by just having tracking by file vs habit.).

<img width="701" alt="harsh_go_—_harsh" src="https://user-images.githubusercontent.com/2866/126034723-26c8b1cf-2bae-427e-afad-284ccd58c974.png">

**Comment by wakatara** [comment]
If you are using homebrew or snaps (or just go install... ) you can now have this feature live in:

https://github.com/wakatara/harsh/releases/tag/v0.8.10

Interested in feedback. Not sure if it's what you want, but would love to hear if it (mostly) fits your needs.

Feel free to comment to reopen the ticket.

**Comment by minthemiddle** [comment]
@wakatara Thanks, tried it out. Exactly what I was looking for. Merci!

**Comment by wakatara** [comment]
@minthemiddle 

Awesome. Glad it works for you.

Out of curiosity, I briefly considered calculating *number* of breaks and skips (esp important when people have say non-weekly or monthly+ tasks). What would you consider more useful for you, the days metric, or the number? (since I keep thinking says is less useful for me than number of times I've broken the chain or skipped. 

lemme know!
Daryl.

**Comment by minthemiddle** [comment]
@wakatara For me, rather days as I can get a better feeling for the overall completion rate.
